sysbuild/Kconfig.suit

```diff
@@ -69,11 +69,22 @@ endif # SUIT_ENVELOPE
 
 config SUIT_BUILD_RECOVERY
 	bool "Build SUIT recovery firmware"
-	depends on NETCORE_HCI_IPC
 	help
 	  Build the recovery firmware, which allows to recover the main application
 	  if the image on the board was damaged.
 
+if SUIT_BUILD_RECOVERY
+
+config SUIT_BUILD_RECOVERY_FIRMWARE_PATH
+	string "Path to the recovery firmware source directory"
+	default "${ZEPHYR_NRF_MODULE_DIR}/samples/suit/recovery"
+	help
+	  This path must contain an application which can be used to recover the main application.
+	  The directory must contain a recovery.cmake file, which allows to build the recovery
+	  application from the main application build and fills in variables holding the 
```

Comment:
no, this is not how it should be done, not at all, please look at how the netcore image is added or how the application core it added and follow that model

Reply:
There is no possibility to do this in the "proper way" for the recovery application after sysbuild has been introduced. The recovery image is _not_ a standard subimage. Building it requires many processes and generating SUIT data which is not generated for standard image.

I have already asked you how to solve the issues with recovery some time ago, but as it was on MSTeams I am pasting the explanation here: 

What we are trying to achieve is that we would like to have an almost "plug and play" recovery application for the user, which the user could connect to his own application. The default version is a dual-core recovery application (application+radiocore). The radiocore image is a standard hci_ipc image, however with many unneeded Kconfig options turned off to reduce memory usage.
 
The only thing which the user _should_ pass to the recovery application from his application is hardware/memory configuration (.overlay files) - this should be passed to both the application and radio cores. However, in general, he should _not_ pass the Kconfig configuration (.conf) files to the recovery application, as these will be chosen by us to highly optimize the recovery application for memory usage.

So basically the recovery application is almost a completely separate project, however it needs some data passed from the main application. 

The problems which building with sysbuild causes are:
 
1) When adding the radio core image (hci_ipc) from sysbuild.cmake in the recovery application, all the configuration of hci_ipc from the recovery_application is ignored - instead the configuration from the top level application is applied. As the recovery app should be "plug and play", this configuration should be managed by the Nordic provided recovery app. I've tried to work around this by temporarly changing the `APP_DIR`, but this causes the problem, that in this case we won't be able to pass the .overlay files from the top level application to the radio core (which we need) + it does not seem to be a safe long term solution.
 
2) The second aspect is passing the overlay files - when passing the .overlay files from the top level application to the recovery application we have two ways: we can add something like `sysbuild/recovery_app.overlay`, but in this case this will only work for a single hardware configuration/board. On the other hand, we can add a sysbuild/recovery_app directory, but this will cause the recovery_app to take all the configuration from the top level directory which we would like to avoid.

---------------

What we want to additionally enable by this PR is giving the user the possibility to provide his own recovery application -  especially, a key client needs the application to work with USB. So we simply decided to give the user as much freedom as possible, allowing him to choose what he wants to include from the application or not.
We see too many scenarios to make a generic solution.

Reply:
To be discussed further during the planned meeting.